### PR TITLE
[risk=low][RW-11801] Dataset export UI update for RStudio

### PR DESCRIPTION
**Pull request by jmthibault79**
Partial completion of RW-11801:
* Whenever a notebook is mentioned, we now specify that we mean a **Jupyter** notebook
* Text is added explaining that direct export to notebooks is only for Jupyter
* Language choice is now first

Screenshot 2024-03-11 at 1 00 58 PM

The "disable for SAS" portion of RW-11801 will be done along with SAS support in #8424

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [x] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/data/data-set/export-dataset-modal.tsx [hunk 1]
```diff
@@ -357,30 +360,12 @@ export const ExportDatasetModal = ({
         <div style={{ width: 'calc(450px - 3rem)' }}>
           <ModalTitle>Export Dataset</ModalTitle>
           <ModalBody>
-            <div style={{ marginTop: '1.5rem' }}>
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-dataset-modal.tsx:360]: moved below "Select programming language"

File: ui/src/app/pages/data/data-set/export-dataset-modal.tsx [hunk 2]
```diff
@@ -406,6 +391,31 @@ export const ExportDatasetModal = ({
                 </label>
               ))}
 
+            <div style={{ marginTop: '1.5rem' }}>
+              <Select
+                isDisabled={shouldDisable}
+                value={creatingNewNotebook ? '' : notebookNameWithoutSuffix}
+                data-test-id='select-notebook'
+                options={selectOptions}
+                onChange={onNotebookSelect}
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-dataset-modal.tsx:400]: slight cleanup here

File: ui/src/app/pages/data/data-set/export-dataset-modal.tsx [hunk 3]
```diff
@@ -406,6 +391,31 @@ export const ExportDatasetModal = ({
...
+              />
+            </div>
+
+            {creatingNewNotebook && (
+              <label>
+                <SmallHeader style={{ fontSize: 14, marginTop: '1.5rem' }}>
+                  Jupyter Notebook Name
+                </SmallHeader>
+                <TextInput
+                  onChange={setNotebookNameWithoutSuffix}
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-dataset-modal.tsx:410]: slight cleanup here.  also added `onBlur` which is required but was missing

**Review by evrii — APPROVED** [review]
Looks good to me. Thanks for taking care of this.

**Pull request merged**